apps/emqx/src/emqx_tls_lib.erl

```diff
@@ -591,71 +592,106 @@ do_drop_invalid_certs([KeyPath | KeyPaths], SSL) ->
 
 %% @doc Convert hocon-checked ssl server options (map()) to
 %% proplist accepted by ssl library.
--spec to_server_opts(tls | dtls, map()) -> [{atom(), term()}].
+-spec to_server_opts(tls | dtls, map()) -> [ssl:tls_server_option()].
 to_server_opts(Type, Opts) ->
-    Versions = integral_versions(Type, maps:get(versions, Opts, undefined)),
-    Ciphers = integral_ciphers(Versions, maps:get(ciphers, Opts, undefined)),
-    Path = fun(Key) -> resolve_cert_path_for_read_strict(maps:get(Key, Opts, undefined)) end,
-    Password = ensure_password(maps:get(password, Opts, undefined)),
-    ensure_valid_options(
-        maps:to_list(Opts#{
-            keyfile => Path(keyfile),
-            certfile => Path(certfile),
-            cacertfile => Path(cacertfile),
-            ciphers => Ciphers,
-            versions => Versions,
-            password => Password
-        }),
-        Versions
-    ).
+    Versions = integral_versions(Type, conf_get_opt(versions, Opts)),
+    Ciphers = integral_ciphers(Versions, conf_get_opt(ciphers, Opts)),
+    DefaultUserLookupFun =
+        case Versions of
+            ['tlsv1.3'] -> undefined;
+            [_ | _] -> {fun emqx_tls_psk:lookup/3, undefined}
+        end,
+    TLSServerOpts = [
+        {versions, Versions},
+        {ciphers, Ciphers}
+        | emqx_utils:flattermap(
```

Comment:
Should we crash if an unknown option is found?
To avoid a situation where an option is added to the hocon schema, but this converter was not updated?

Reply:
Good question. First thing to note is that `ocsp` option is already unknown in some sense (but we're aware of it). I'd argue that introduction of new schema fields should be paired with relevant feature tests. But, if it's for some reason not, and the field in question is optional, this "crash if unknown" logic would introduce a potential risk crashing listeners in runtime later, on someone's production system.